### PR TITLE
Batch execution refactor

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

To help fix a bug that returns phantom events.

### What changes were made as part of this PR

- refactor the "batch executor" logic to make it more clear
- add a fk from `receipt.tx` to the transaction table, because now we 
- add a wait for the validator to return the RPC encryption key

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/components/batch_executor.go [hunk 1]
```diff
@@ -210,219 +241,188 @@ func (executor *batchExecutor) ComputeBatch(ctx context.Context, context *BatchE
...
+	if err != nil {
+		return fmt.Errorf("could not process on block end transaction hook. Cause: %w", err)
+	}
+	// Ensure the onBlock callback transaction is successful. It should NEVER fail.
+	if err = executor.verifySyntheticTransactionsSuccess(onBlockPricedTx, onBlockTxResult); err != nil {
+		return fmt.Errorf("batch computation failed due to onBlock hook reverting. Cause: %w", err)
+	}
+	ec.blockEndResult = onBlockTxResult
+	ec.blockEndResult.MarkSynthetic(true)
+	// todo - stefan not clear what this does
```
Comment (StefanIliev545) [inline go/enclave/components/batch_executor.go:361]: Some paranoia checks I've added; Not sure if we really need them

File: go/enclave/storage/enclavedb/batch.go [hunk 2]
```diff
@@ -79,10 +79,10 @@ func WriteTransactions(ctx context.Context, dbtx *sql.Tx, transactions []*core.T
 
 			args = append(args, transaction.Tx.Hash()) // tx_hash
 			args = append(args, txBytes)               // content
-			args = append(args, toContracts[i])        // To
+			args = append(args, toContractIds[i])      // To
 			args = append(args, transaction.Tx.Type()) // Type
-			args = append(args, senders[i])            // sender_address
-			args = append(args, i)                     // idx
+			args = append(args, senderIds[i])          // sender_address
+			args = append(args, fromIdx+i)             // idx
```
Comment (StefanIliev545) [inline go/enclave/storage/enclavedb/batch.go:85]: I take it this was wrong value previously

**Review by StefanIliev545 — APPROVED** [review]
LGTM, dont see anything wrong;

File: go/enclave/storage/enclavedb/batch.go (hunk 2 above)
Comment (tudor-malene) [reply go/enclave/storage/enclavedb/batch.go:85]: I'm not sure about the offsets that were passed to the tx execution either

**Pull request merged**